File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnPlaylistIdentifier.java

package com.gracenote.gnsdk;

/** 
* <b>Experimental</b>: {@link GnPlaylistIdentifier} 
*/ 
 
public class GnPlaylistIdentifier {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnPlaylistIdentifier(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnPlaylistIdentifier obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnPlaylistIdentifier(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public String mediaIdentifier() {
    return gnsdk_javaJNI.GnPlaylistIdentifier_mediaIdentifier(swigCPtr, this);
  }

  public String collectionName() {
    return gnsdk_javaJNI.GnPlaylistIdentifier_collectionName(swigCPtr, this);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnPlaylistIdentifier.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* <b>Experimental</b>: GnPlaylistIdentifier
*/
public class GnPlaylistIdentifier : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnPlaylistIdentifier(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnPlaylistIdentifier obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnPlaylistIdentifier() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnPlaylistIdentifier(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

  public string MediaIdentifier {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistIdentifier_MediaIdentifier_get(swigCPtr) );
	} 

  }

  public string CollectionName {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistIdentifier_CollectionName_get(swigCPtr) );
	} 

  }

}

}
